tests/cmsis_rtos_v2/src/thread_apis.c

```diff
@@ -29,6 +30,80 @@ static osThreadAttr_t thread2_attr = {
 	.priority = osPriorityHigh,
 };
 
+static void thread1_dynamic(void *argument)
```

Comment:
The body of the newly introduced functions are almost same as the existing functions. It is better to use the existing functions itself and distinguish between dynamic and regular threads using arguments taken by the thread functions.

Reply:
Sure - I'll pass pointers to the globals in as thread arguments.